### PR TITLE
Use arange_like in AutoRegressiveBias block to reduce overall number of ops

### PR DESCRIPTION
**Pull request by fhieber**
Addition of `arange_like` since MXNet 1.6 allows us to simplify AutoRegressiveBias a little bit.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] You have considered writing a test
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [x] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sockeye/transformer.py [hunk 1]
```diff
@@ -381,15 +381,12 @@ def cast(self, dtype):
...
-        length_array = F.cast(F.contrib.index_array(x, axes=(1,)), dtype=self._dtype)
+        # Shape: (length, 1)
+        length_array = F.contrib.arange_like(x, axis=1)
         # matrix with lower triangle and main diagonal set to 0, upper triangle set to 1
         # Shape: (length, length)
-        bias = F.broadcast_greater(F.reshape(length_array, shape=(1, -1)),
-                                   length_array)
+        bias = F.broadcast_greater(F.expand_dims(length_array, axis=0),
+                                   F.expand_dims(length_array, axis=1))
         bias = bias * -C.LARGE_VALUES[self._dtype]
```
Comment (barbaradarques) [inline sockeye/transformer.py:390]: [Curiosity only] Why is this necessary? I read the docs and saw that for np.float16 it will move it to some multiple of 32 and that for np.float32 it will move to 1.0e8. Is the goal to keep the np.float32 masked logits intact and round them up when using np.float16?

**Review by barbaradarques — APPROVED** [review]

File: sockeye/transformer.py (hunk 1 above)
Comment (fhieber) [reply sockeye/transformer.py:390]: If training in FP16 you want to avoid overflowing the numerical range, therefore the max value for FP16 is different than for FP32.
